t/002.upstream_dynamic.t

```diff
@@ -116,3 +118,55 @@ location = / {
 [200, 200, 999, 999, 200, 200, 999, 999]
 --- response_body eval
 ["Pass", "Pass", "Backup", "Backup", "Pass", "Pass", "Backup", "Backup"]
+=== TEST 5: Dynamic upstream with constantly timeout DNS resolution without strict fallback
+--- init
+`echo 'local-data: "example.com 1 A 127.0.0.2"' > /tmp/unbound_local_zone_ngx_upstream_jdomain.conf && unbound-control reload ` or die $!;
+--- http_config
+resolver 127.0.0.88;
+upstream upstream_test {
+	jdomain example.com port=8000 retry_off fallback=127.0.0.3:12345 ;
+}
+server {
+	listen 127.0.0.3:12345;
+	return 999 'Backup';
+}
+server {
+	listen 127.0.0.2:8000;
+	return 200 'Kill';
+}
+--- config
+location = / {
+	proxy_pass http://upstream_test;
+}
+--- request eval
+["GET /", "GET /", "GET /", "GET /", "GET /", "GET /", "GET /", "GET /"]
+--- error_code eval
+[200, 200, 200, 200, 200, 200, 200, 200]
+--- response_body eval
+["Kill", "Kill", "Kill", "Kill", "Kill", "Kill", "Kill", "Kill"]
+=== TEST 6: Dynamic upstream with constantly timeout DNS resolution without strict fallback
+--- init
+`echo 'local-data: "example.com 1 A 127.0.0.2"' > /tmp/unbound_local_zone_ngx_upstream_jdomain.conf && unbound-control start ` or die $!;
+--- http_config
+resolver 127.0.0.88;
+upstream upstream_test {
+	jdomain example.com port=8000 retry_off fallback=127.0.0.3:12345 strict;
+}
+server {
+	listen 127.0.0.3:12345;
+	return 999 'Backup';
+}
+server {
+	listen 127.0.0.2:8000;
+	return 200 'Kill';
+}
+--- config
+location = / {
+	proxy_pass http://upstream_test;
+}
+--- request eval
+["GET /", "GET /", "GET /", "GET /", "GET /", "GET /", "GET /", "GET /"]
+--- error_code eval
+[200, 200, 999, 999, 999, 999, 999, 999]
+--- response_body eval
+["Kill", "Kill", "Backup", "Backup", "Backup", "Backup", "Backup", "Backup"]
```

Comment:
With the update above, not killing unbound server, the perl script is able to recover the state, so these would alternate again as expected:

```suggestion
--- error_code eval
[200, 200, 999, 999, 200, 200, 999, 999]
--- response_body eval
["Kill", "Kill", "Backup", "Backup", "Kill", "Kill", "Backup", "Backup"]
```